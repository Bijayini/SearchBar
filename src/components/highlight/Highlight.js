import React from 'react';
import PropTypes from "prop-types";

const escapeRegExp = (str) => {
  return str.replace(/[-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&");
};

const Highlighted = ({text, highlight, style}) => {
  if (!highlight.trim()) {
    return <span>{text}</span>
  }
  const regex = new RegExp(`(${escapeRegExp(highlight)})`, 'gi');
  const parts = text.split(regex);

  return (
    <p style={style}>
        {parts.filter(part => part).map((part, i) => (
          regex.test(part) ? <mark key={i}>{part}</mark> : <span key={i}>{part}</span>
        ))}
    </p>
  )
};

Highlighted.defaultProps = {
  style:null,
  text:'',
  highlight:''
};

Highlighted.propTypes = {
  style:PropTypes.object,
  text:PropTypes.string,
  highlight:PropTypes.string
};

export default Highlighted;
