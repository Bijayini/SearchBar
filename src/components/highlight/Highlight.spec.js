import React from "react";
import { render } from "@testing-library/react";
import Highlight from "./Highlight";

describe("Highlight component", () => {
  const props = {
    text:'some data',
    highlight:'So',
    style:{
      fontWeight:700
    }
  };

  it("should render Highlight when highlight is present", () => {
    const {baseElement} = render(<Highlight {...props} />);

    expect(baseElement).toMatchSnapshot();
  });

  it("should not have mark tag when highlight is empty", () => {
    const newProps = {
      ...props,
      highlight: ''
    };
    const {baseElement} = render(<Highlight {...newProps} />);

    expect(baseElement.querySelector('mark')).not.toBeInTheDocument()
  });

});
