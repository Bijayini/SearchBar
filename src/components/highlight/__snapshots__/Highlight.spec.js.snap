// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight component should render Highlight when highlight is present 1`] = `
<body>
  <div>
    <p
      style="font-weight: 700;"
    >
      <mark>
        so
      </mark>
      <span>
        me data
      </span>
    </p>
  </div>
</body>
`;
